fieldpath/path_test.go

```diff
@@ -31,16 +38,16 @@ func TestPathString(t *testing.T) {
 		{"basic1", MakePathOrDie("foo", 1), ".foo[1]"},
 		{"basic2", MakePathOrDie("foo", "bar", 1, "baz"), ".foo.bar[1].baz"},
 		{"associative-list-ref", MakePathOrDie("foo", KeyByFields(
-			"a", value.StringValue("b"),
-			"c", value.IntValue(1),
-			"d", value.FloatValue(1.5),
-			"e", value.BooleanValue(true),
+			"a", "b",
+			"c", 1,
+			"d", 1.5,
```

Comment:
This reads better now, but it's no longer obvious that we're testing all the types. Maybe a comment would help?